### PR TITLE
Update user.php

### PR DESCRIPTION
**Pull request by pemaier**
Updated getUsers and getTotalUsers / added filter entries

### CODE REVIEW

File: upload/admin/model/user/user.php [hunk 1]
```diff
@@ -175,7 +217,47 @@ public function getUsers(array $data = []): array {
 	 * @return int
 	 */
 	public function getTotalUsers(): int {
-		$query = $this->db->query("SELECT COUNT(*) AS `total` FROM `" . DB_PREFIX . "user`");
+		$sql = "SELECT COUNT(*) AS `total` FROM `" . DB_PREFIX . "user` `u` ";
+
+		$implode = [];
+
+		if (!empty($data['filter_username'])) {
+			$implode[] = "LCASE(`u`.`username`) LIKE '" . $this->db->escape(oc_strtolower($data['filter_username']) . '%') . "'";
```
Comment (AJenbo) [inline upload/admin/model/user/user.php:225]: This function does not have a $data variable so you need to add it, maybe via a function argument like it is for `getUsers()`.

```php
public function getTotalUsers(array $data = []): int {
```

**Comment by pemaier** [comment]
@AJenbo Thank you for the hint. Pls. be so kind and add it ... as I think you have the rights to do. Thank you for your support.

**Comment by AJenbo** [comment]
@pemaier for the record you as the PR author also have the right to change files in it.

image

I also noticed that you appear to be creating multiple PRs for a single feature with just one file changed in each PR. It would be much better if you could change all the necessary files in the same PR as they can then be reviewed, tested and merged together.

Normally you would do the changes locally on your computer and then push them to a branch on your fork, but it appears that you mainly do changes directly in the browser. It's still possible to edit multiple files in a single PR from the web interface, I think the easiest might be to go to your initial PR and then press `.` this will open the web version of Visual Code.

image

Here you can go to the file manager and navigate and edit the files you need, then go to the Source Control section and to stage, commit and push your changes.

**Comment by pemaier** [comment]
@AJenbo Thank you :+1: :))

**Pull request merged**